### PR TITLE
Merge nuspec and CommonAssemblyInfo into csproj

### PR DESCRIPTION
**Pull request by scott-xu**
Suggest using [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning) after this PR

### CODE REVIEW

File: src/Renci.SshNet/Renci.SshNet.csproj [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
-    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <AssemblyName>Renci.SshNet</AssemblyName>
-	<TargetFrameworks>net462;netstandard2.0;netstandard2.1;net6.0;net7.0;net8.0</TargetFrameworks>
+    <Product>SSH.NET</Product>
+    <AssemblyTitle>SSH.NET</AssemblyTitle>
+    <TargetFrameworks>net462;netstandard2.0;netstandard2.1;net6.0;net7.0;net8.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
Comment (Rob-Hague) [inline src/Renci.SshNet/Renci.SshNet.csproj]: This should not be in the csproj (except for testing). We should use `dotnet pack` when we need to generate a package

```suggestion
```

**Comment by Rob-Hague** [comment]
Will you also delete https://github.com/sshnet/SSH.NET/blob/develop/build/nuget/SSH.NET.nuspec?

(assuming this works well with sandcastle etc.)

**Comment by scott-xu** [comment]
> Will you also delete https://github.com/sshnet/SSH.NET/blob/develop/build/nuget/SSH.NET.nuspec?
> 
> (assuming this works well with sandcastle etc.)

I didn't delete the nuspec because I need confirmation from @WojciechNagorski.

**Comment by scott-xu** [comment]
Regarding sandcastle, it would be good to migrate to [docfx](https://dotnet.github.io/docfx/index.html)

File: src/Renci.SshNet/Renci.SshNet.csproj [hunk 2]
```diff
@@ -1,23 +1,52 @@
...
+    <AssemblyTitle>SSH.NET</AssemblyTitle>
+    <TargetFrameworks>net462;netstandard2.0;netstandard2.1;net6.0;net7.0;net8.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <PackageId>SSH.NET</PackageId>
+    <Title>SSH.NET</Title>
+    <Version>2023.0.1</Version>
+    <Description>SSH.NET is a Secure Shell (SSH) library for .NET, optimized for parallelism.</Description>
+    <Copyright>Copyright © Renci 2010-2023</Copyright>
```
Comment (drieseng) [inline src/Renci.SshNet/Renci.SshNet.csproj]: Can we replace **2023** with a dynamic value?

Comment (scott-xu) [reply src/Renci.SshNet/Renci.SshNet.csproj]: This PR only focus on the merging. Versioning could be another PR.

Comment (Rob-Hague) [reply src/Renci.SshNet/Renci.SshNet.csproj]:
```suggestion
    <Copyright>Copyright © Renci 2010-$([System.DateTime]::UtcNow.Year)</Copyright>
```

**Comment by scott-xu** [comment]
> > Will you also delete https://github.com/sshnet/SSH.NET/blob/develop/build/nuget/SSH.NET.nuspec?
> > (assuming this works well with sandcastle etc.)
> 
> I didn't delete the nuspec because I need confirmation from @WojciechNagorski.

@WojciechNagorski @drieseng , what do you think? Should we delete SSH.NET.nuspec?

**Comment by WojciechNagorski** [comment]
@scott-xu I'm sorry that the review is delayed, but I need more time to think and review it thoroughly.

**Comment by Rob-Hague** [comment]
It looks good. Before (using build/build.proj per the [wiki](https://github.com/sshnet/SSH.NET/wiki/Build-a-new-release)):

<details>

image

</details>

After (using `dotnet pack`):

<details>

image

</details>

I think the next steps are:

1. Delete the nuspec
2. Modify build.proj to use `dotnet pack` instead of nuget.exe

Would it be possible to do these in this PR? That way we don't end up with two different ways to generate the release artifacts, and the new nuget method should integrate seamlessly into the current release process (which will also help with validation).

After that (in another PR) maybe we can call build.proj in CI to generate the artifacts.

**Comment by WojciechNagorski** [comment]
I will try to find time next week for this PR

**Comment by WojciechNagorski** [comment]
Hi I'm busy but I still remember about this PR.

**Review by Rob-Hague — APPROVED** [review]
Nice. Now we have source-link 🙂 

`msbuild /p:ReleaseVersion=2023.0.1 /t:Package`:

image

**Review by WojciechNagorski — APPROVED** [review]

**Comment by WojciechNagorski** [comment]
This issue has been fixed in the 2024.0.0 version.

**Pull request merged**